File: Cpp/Number of 1 Bits.cpp
// 6ms, 2.03%
class Solution
{
public:
    int hammingWeight(uint32_t n)
    {
        int res = 0;
        while(n)
        {
            n &= n - 1;
            ++res;
        }
        return res;
    }
};

File: Python3/Number of 1 Bits.py
# 36ms, 45.97%
class Solution:
    def hammingWeight(self, n):
        count = 0
        while n:
            n &= n - 1
            count += 1
        return count
        '''
        ### 37ms, 45.97%
        count = 0
        while n:
            count += n & 1
            n >>= 1
        return count
        '''
